setup.py

```diff
@@ -38,6 +38,13 @@ class Components:
     )
 
     __version__ = v[: v.rfind("-")].replace("-", ".") if "-" in v else v
+
+    # if version is not a valid PEP440 version, then create a bogus version
+    # that will be used only for development purposes. this usually happens
+    # when we do a shallow clone of the repo (e.g. in github actions)
+    if not re.match(r"^\d+(\.\d+)*$", __version__):
+        __version__ = "0.0.dev0"
```

Comment:
I don't see anything wrong with this and if it overcomes a difficulty, then great. I take it that encoding a bit of the sha hash into the __version__ (e.g. in the dev0 part) would not be PEP440 compliant or not provide any extra usefulness?

Reply:
It might come in handy, I'll add it